### PR TITLE
LoongArch: purge bash-ism and intrusive macro definitions.

### PR DESCRIPTION
**Pull request by scylaac**
https://gcc.gnu.org/pipermail/gcc-patches/2022-March/591253.html

### CODE REVIEW

**Comment by scylaac** [comment]
bot test this please

File: gcc/config/loongarch/loongarch.cc [hunk 1]
```diff
@@ -4090,6 +4090,21 @@ loongarch_expand_block_move (rtx dest, rtx src, rtx length)
...
+
+bool
+loongarch_do_optimize_block_move_p (void)
+{
+  if (target_flags_explicit & MASK_MEMCPY)
+    /* if -m[no-]memcpy is given explicitly */
+    return !TARGET_MEMCPY;
+  else
+    /* if not, don't optimize under -Os */
+    return !optimize_size;
```
Comment (xen0n) [inline gcc/config/loongarch/loongarch.cc:4104]: `else` is unneeded because the "then" branch is an early return.

**Comment by scylaac** [comment]
bot test this please

**Review by xen0n — APPROVED** [review]
I didn't test the changes myself but it looks mostly valid.

**Pull request merged**